### PR TITLE
Add search to JobModal.tsx

### PR DESCRIPTION
**Pull request by and189**
add search

### CODE REVIEW

File: packages/client/src/app/jobs/executeJobModal.tsx [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
-            throw new Error();
-          }
+  const filteredDevices = useMemo(() => {
+    return devices?.filter((device) => device.deviceId?.includes(search) || device.origin?.includes(search)) || [];
+  }, [devices, search]);
 
-          closeModal();
-        },
-      );
+  const executeJob = async ({ deviceIds }: { deviceIds: string[] | number[] }) => {
```
Comment (na-ji) [inline packages/client/src/app/jobs/executeJobModal.tsx]: Why did you remove the useCallback here? 🤔

File: packages/client/src/app/jobs/executeJobModal.tsx [hunk 2]
```diff
@@ -20,19 +27,17 @@ export const ExecuteJobModal = ({ closeModal, devices, jobId }: ExecuteJobModalP
...
       toast.promise(promise, {
         pending: `Running ${jobId}...`,
-        success: `${jobId} started succesfully`,
+        success: `${jobId} started successfully`,
         error: `Failed to start job ${jobId}`,
       });
     },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [jobId],
+    [jobId, closeModal], // Abhängigkeiten
```
Comment (na-ji) [inline packages/client/src/app/jobs/executeJobModal.tsx]: What does this comment mean?

Comment (Fabio1988) [reply packages/client/src/app/jobs/executeJobModal.tsx]: Dependencies, German word. Comments should be English though :)

Comment (lenisko) [reply packages/client/src/app/jobs/executeJobModal.tsx]: lol, in a future... use English in all the code you write, unless it's a "dupa" then it's fine.

**Review by na-ji — APPROVED** [review]
Thanks for the PR

File: packages/client/src/app/jobs/executeJobModal.tsx [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
 }
 
-export const ExecuteJobModal = ({ closeModal, devices, jobId }: ExecuteJobModalProps): JSX.Element => {
+export const ExecuteJobModal: React.FC<ExecuteJobModalProps> = ({ closeModal, devices, jobId }) => {
   const [selectedDevices, setSelectedDevices] = useState<Selection>();
+  const [search, setSearch] = useState('');
+
+  const filteredDevices = useMemo(() => {
+    return devices?.filter((device) => device.deviceId?.includes(search) || device.origin?.includes(search)) || [];
+  }, [devices, search]);
```
Comment (na-ji) [inline packages/client/src/app/jobs/executeJobModal.tsx]: Just tested it, and there is a small issue: it's sensible to the case. It could be cool if it wasn't. One of the way to do it:
```suggestion
  const filteredDevices = useMemo(() => {
    return (
      devices?.filter(
        (device) =>
          device.deviceId?.toLowerCase().includes(search.toLowerCase()) ||
          device.origin?.toLowerCase().includes(search.toLowerCase()),
      ) || []
    );
  }, [devices, search]);
```
You could also prevent transforming multiple times to lower case the same variable by saving the lower case value to an intermediary value.

File: packages/client/src/app/jobs/executeJobModal.tsx [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
   devices?: MitmControlDTO[];
   jobId: string;
 }
 
-export const ExecuteJobModal = ({ closeModal, devices, jobId }: ExecuteJobModalProps): JSX.Element => {
+export const ExecuteJobModal: React.FC<ExecuteJobModalProps> = ({ closeModal, devices, jobId }) => {
   const [selectedDevices, setSelectedDevices] = useState<Selection>();
+  const [search, setSearch] = useState('');
+
+  const filteredDevices = useMemo(() => {
```
Comment (Fabio1988) [inline packages/client/src/app/jobs/executeJobModal.tsx:19]: @na-ji I am not the best in React, but do we always build filteredDevices,even the search is empty?

Comment (na-ji) [reply packages/client/src/app/jobs/executeJobModal.tsx:19]: Here, the hook useMemo prevent this. Basically, when doing the first render, it will compute the filtered list even if the search query is empty. On subsequent renders, it will only compute this array again if the search was updated.

**Comment by na-ji** [comment]
Thanks for the PR
